uint8 num_bin
uint16[] xData 

#origin y value
uint16[] yData

uint16 yLowThreshold
uint16 yHighThreshold

#binary y value
uint16[] yBinData